giskard_cicd/automation/post_discussion.py

```diff
@@ -73,37 +74,52 @@ def save_post(report_path, path, dataset_id, dataset_config, dataset_split):
     with open(path, "w") as f:
         f.write(post)
 
-
-def separate_report_by_issues(report):
-    # TODO: add markdown comments to the report as a split marker
-    regex = (
-        "\W(?="
-        + "|".join(["<details>\n<summary>👉" + issue for issue in ISSUE_GROUPS])
-        + ")"
-    )
-    sub_reports = re.split(regex, report)
-    return sub_reports
-
+class Issue:
+    def __init__(self, description, examples):
+        self.description = description
+        self.examples = examples
+    
+    def __len__(self):
+        return len(self.description) + len(self.examples)
+    
+    def trim_examples(self):
+        # get characters count of the examples
+        if len(self.examples) > 60000:
+            self.examples = "</details>"
+
+def load_report_to_issues(report):
+    splited_issues = []
+    # <!-- issue --> is used to separate the issues
+    issues = [ issue for issue in report.split("<!-- issue -->") if len(issue) > 0 ]
+    # <!-- examples --> is used to separate the examples
+    for issue in issues:
+      splited_issue = issue.split("<!-- examples -->")
+      description = splited_issue[0]
+      examples = splited_issue[1]
+      splited_issues.append(Issue(description, examples))
+    return splited_issues
 
 def post_issue_as_comment(discussion, issue, token, repo_id):
-    comment = hf_hub.comment_discussion(
-        repo_id=repo_id,
-        repo_type="space",
-        discussion_num=discussion.num,
-        comment=issue,
-        token=token,
-    )
-    return comment
+    try:
```

Comment:
I think we can have this try..except first in cicd through a separated PR, before delivering https://github.com/Giskard-AI/giskard/pull/1812 in the new giskard release. WDYT?

Reply:
NVM, we will have the release sooner today